sdk/core/core/README.md

```diff
@@ -8,182 +8,183 @@ The library allows client libraries to expose common functionality in a consiste
 
-### Retry policy
+To enable logging, you must first write a callback function that our logging mechanism will invoke periodically with messages. The function signature must match this type definition (defined in the [az_log.h](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/core/core/inc/az_log.h) file):
 
-While working with Azure you might encounter transient failures. For information about handling these types of failures, see [Retry pattern][azure_pattern_retry] in the Cloud Design Patterns guide, and the related [Circuit Breaker pattern][azure_pattern_circuit_breaker].
+   ```C
+   typedef void (*az_log_fn)(az_log_classification classification, az_span message);
+   ```
 
-## Next steps
+And then, during your application’s initialization, you must register your function with our SDK by calling this function:
 
-### More sample code
+   ```C
+   void az_log_set_listener(az_log_fn listener);
+   ```
 
-TODO
+Now, whenever our SDK wants to send a log message, it will invoke your callback function passing it the log classification and an `az_span` containing the message string (not 0-terminated). Your callback method can now do whatever it wants to with this message such as append it to a file or write it to the console. 
 
-### Additional documentation
+**Note:** In a multi-threaded application, multiple threads may invoke this callback function simultaneously; if your function requires any kind of thread synchronization, then you must provide it.
 
-TODO
+Log classifications allow your application to select which specific log messages it wants to receive. Here is a complete example that logs HTTP request and response messages to standard output:
+
+   ```C
+   void test_log_func(az_log_classification classification, az_span message)
+   {
+      printf("%.*s\n", az_span_length(message), az_span_ptr(message));
+   }
+
+   int main()
+   {
+      az_log_classification const classifications[] = { AZ_LOG_HTTP_REQUEST, AZ_LOG_HTTP_RESPONSE, AZ_LOG_END_OF_LIST };
+      az_log_set_classifications(classifications);
+      az_log_set_callback(test_log_func);
 
-## Contributing
-For details on contributing to this repository, see the [contributing guide][azure_sdk_for_c_contributing].
+      // More code goes here...
+   }
+   ```
 
-This project welcomes contributions and suggestions.  Most contributions require you to agree to a
-Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us
-the rights to use your contribution. For details, visit https://cla.microsoft.com.
+### SDK Function Argument Validation
 
-When you submit a pull request, a CLA-bot will automatically determine whether you need to provide
-a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions
-provided by the bot. You will only need to do this once across all repos using our CLA.
+Public SDK functions validate the arguments passed to them in an effort to ensure that calling code is 
```

Comment:
Markdown linter generally doesn't like the extra trailing white space/explicit line breaks. Removing those.